<style>

* {
    font-family: Cambria, serif;
font-size: 18px;
font-style: normal;
font-weight: normal;
text-transform: normal;
letter-spacing: normal;
line-height: 1.4em;
}


div.article {padding: 3px; border: 3px solid #C3D9FF; }
div.article_title { color:#356AA0; font-weight: bold;}
div.article_abstract {padding-top: 4px;}

span.article_year {border-right: 10px; color: #D15600;}

</style>

<table>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Localization of mouse satellite DNA in constitutive heterochromatin.</div>
   <div class="article_authors">
    <span class="article_year">1970</span>
    
      <span class="article_author">Yasmineh WG</span>
    
      <span class="article_author">Yunis JJ</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Long range periodicities in mouse satellite DNA.</div>
   <div class="article_authors">
    <span class="article_year">1975</span>
    
      <span class="article_author">Southern EM</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Characterization of distinct segments in mouse satellite DNA by restriction nucleases.</div>
   <div class="article_authors">
    <span class="article_year">1977</span>
    
      <span class="article_author">Horz W</span>
    
      <span class="article_author">Zachau HG</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">The long-range periodicity of mouse satellite DNA has been analyzed by digestion with five restriction nucleases. With all nucleases tested, a major repeat unit approximately 245 nucleotide pairs became apparent. Minor registers of shorter length were also detected. The total number of cleavage sites per haploid genome for each restriction enzyme as well as their positions relative to each other were determined. While endo R-EcoRII was known to cleave all of the satellite DNA, the other four restriction enzymes were found to generate only weak degradation patterns. The results taken together with quantitative analyses of codigestion experiments indicate that the recognition sequences for each of these four nucleases are clustered on separate parts of the satellite DNA. It is concluded that the satellite DNA, which appears homogeneous by digestion with endo R-EcRII, contains distinct segments each susceptible to degradation with one of the other nucleases. These results have certain implications for theories on the evolution of mouse satellite DNA. A simple mechanism of multiplication and divergence by mutation is not sufficient to explain the data. Additional and alternative processes which are relevant to the evolutionary considerations are discussed.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Characterization of distinct segments in mouse satellite DNA by restriction nucleases.</div>
   <div class="article_authors">
    <span class="article_year">1977</span>
    
      <span class="article_author">Horz W</span>
    
      <span class="article_author">Zachau HG</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">The long-range periodicity of mouse satellite DNA has been analyzed by digestion with five restriction nucleases. With all nucleases tested, a major repeat unit approximately 245 nucleotide pairs became apparent. Minor registers of shorter length were also detected. The total number of cleavage sites per haploid genome for each restriction enzyme as well as their positions relative to each other were determined. While endo R-EcoRII was known to cleave all of the satellite DNA, the other four restriction enzymes were found to generate only weak degradation patterns. The results taken together with quantitative analyses of codigestion experiments indicate that the recognition sequences for each of these four nucleases are clustered on separate parts of the satellite DNA. It is concluded that the satellite DNA, which appears homogeneous by digestion with endo R-EcRII, contains distinct segments each susceptible to degradation with one of the other nucleases. These results have certain implications for theories on the evolution of mouse satellite DNA. A simple mechanism of multiplication and divergence by mutation is not sufficient to explain the data. Additional and alternative processes which are relevant to the evolutionary considerations are discussed.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">The compaction of mouse heterochromatin as studied by nuclease digestion.</div>
   <div class="article_authors">
    <span class="article_year">1981</span>
    
      <span class="article_author">Horvath P</span>
    
      <span class="article_author">Horz W</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Chromosomal and nuclear location of mouse satellite DNA in individual cells.</div>
   <div class="article_authors">
    <span class="article_year">1970</span>
    
      <span class="article_author">Jones KW</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Localization of mouse satellite DNA in constitutive heterochromatin.</div>
   <div class="article_authors">
    <span class="article_year">1970</span>
    
      <span class="article_author">Yasmineh WG</span>
    
      <span class="article_author">Yunis JJ</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Nucleotide sequence of mouse satellite DNA.</div>
   <div class="article_authors">
    <span class="article_year">1981</span>
    
      <span class="article_author">Horz W</span>
    
      <span class="article_author">Altenburger W</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">The nucleotide sequence of uncloned mouse satellite DNA has been determined by analyzing Sau96I restriction fragments that correspond to the repeat unit of the satellite DNA. An unambiguous sequence of 234 bp has been obtained. The sequence of the first 250 bases from dimeric satellite fragments present in Sau96I limit digests corresponds almost exactly to two tandemly arranged monomer sequences including a complete Sau96I site in the center. This is in agreement with the hypothesis that a low level of divergence which cannot be detected in sequence analyses of uncloned DNA is responsible for the appearance of dimeric fragments. Most of the sequence of the 5% fraction of Sau96 monomers that are susceptible to TaqI has also been determined and has been found to agree completely with the prototype sequence. The monomer sequence is internally repetitious being composed of eight diverged subrepeats. The divergence pattern has interesting implications for theories on the evolution of mouse satellite DNA.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Curvature of mouse satellite DNA and condensation of heterochromatin.</div>
   <div class="article_authors">
    <span class="article_year">1987</span>
    
      <span class="article_author">Radic MZ</span>
    
      <span class="article_author">Lundgren K</span>
    
      <span class="article_author">Hamkalo BA</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Cloned, sequenced mouse satellite DNA exhibits properties characteristic of molecules that possess a stable curvature. Circularly permuted fragments containing the region predicted to bend were used to map the curvature relative to DNA sequence. The altered mobility of these fragments in polyacrylamide gels is reversed when gels are run in the presence of distamycin A, a drug that binds preferentially to AT-rich DNA. Treatment of living mouse cells with this drug dramatically reduces the condensation of centromeric heterochromatin, the exclusive location of satellite sequences. In situ hybridization of satellite probes to extended chromosomes at the electron microscope level shows that satellite does not comprise a single block but is distributed throughout the centromere region. Based on these experiments, we hypothesize that the structure of mouse satellite DNA is an important feature of centromeric heterochromatin condensation.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">A simplified method for preparation of mouse satellite DNA.</div>
   <div class="article_authors">
    <span class="article_year">1977</span>
    
      <span class="article_author">Manuelidis L</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">The organization of the mouse satellite DNA at centromeres.</div>
   <div class="article_authors">
    <span class="article_year">1989</span>
    
      <span class="article_author">Joseph A</span>
    
      <span class="article_author">Mitchell AR</span>
    
      <span class="article_author">Miller OJ</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">The mouse genome contains a major and a minor satellite DNA family of repetitive DNA sequences. The use of 5-azacytidine has allowed us to demonstrate that these satellite DNAs are organized in two separate domains at the centromeres of mouse chromosomes. The minor satellite is closer to the short arms of the acrocentric chromosomes than the major satellite. The major satellite is farther away, flanking the minor satellite and adjacent to the euchromatic long arm of each mouse chromosome. At the level of resolution afforded by the in situ hybridization technique it would appear that the organization of the centromeric domain of the mouse is similar to that in man. That is, both contain two repetitive DNA sequence families arranged in major blocks.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Partial sequence analysis of mouse satellite DNA evidence for short range periodicities.</div>
   <div class="article_authors">
    <span class="article_year">1975</span>
    
      <span class="article_author">Biro PA</span>
    
      <span class="article_author">Carr-Brown A</span>
    
      <span class="article_author">Southern EM</span>
    
      <span class="article_author">Walker PM</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Mouse minor satellite DNA genetically maps to the centromere and is physically linked to the proximal telomere.</div>
   <div class="article_authors">
    <span class="article_year">1991</span>
    
      <span class="article_author">Kipling D</span>
    
      <span class="article_author">Ackford HE</span>
    
      <span class="article_author">Taylor BA</span>
    
      <span class="article_author">Cooke HJ</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">As an adjunct to attempts to define functionally important sequences at human centromeres, we have undertaken a long-range physical analysis of these regions in the mouse. Mouse centromeres are usually situated very close to the chromosome ends and are closely associated with minor satellite sequences on the basis of cytological observations. Using pulsed-field gel electrophoresis we find that this satellite DNA is arranged as tandem arrays, predominantly uninterrupted by nonsatellite sequences. These arrays can be released largely intact by digestion with a range of enzymes that generally cleave frequently in non-satellite DNA. The restriction fragments carrying these arrays are polymorphic in size between inbred strains and provide direct markers for mouse centromeres. To illustrate the possible use of these polymorphic markers we have mapped a 1.3-Mb PvuII variant in a set of RI strains to the centromere of Chromosome 7. The minor satellite arrays are very close to the centromeric telomere and physical linkage with terminal repeat sequences can readily be detected, placing many minor satellite arrays on terminal restriction fragments smaller than 1 Mb. The apparent lack of any sizable amount of nonsatellite DNA between the minor satellite and the terminal repeat arrays indicates that many mouse chromosomes are truly telocentric.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Consensus sequence of mouse satellite DNA indicates it is derived from tandem 116 basepair repeats.</div>
   <div class="article_authors">
    <span class="article_year">1981</span>
    
      <span class="article_author">Manuelidis L</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Mouse satellite DMA in noncentromeric heterochromatin of cultured cells.</div>
   <div class="article_authors">
    <span class="article_year">1975</span>
    
      <span class="article_author">White R</span>
    
      <span class="article_author">Pasztor LM</span>
    
      <span class="article_author">Hu F</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Specific chromosomes in several mouse lines have interstitial C-bands. In situ hybridization studies indicate that these interstitial bands contain typical mouse satellite DNA.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Transcripts from opposite strands of gamma satellite DNA are differentially expressed during mouse development.</div>
   <div class="article_authors">
    <span class="article_year">1995</span>
    
      <span class="article_author">Rudert F</span>
    
      <span class="article_author">Bronner S</span>
    
      <span class="article_author">Garnier JM</span>
    
      <span class="article_author">Dolle P</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Using in vitro immuno-selected retinoic acid response elements, we have isolated mouse genomic clones containing major (gamma) satellite DNA repeats that are considered as typical of chromosome centromeres. Several cDNA clones were then isolated from a F9 cell cDNA library and were found to harbor variants of the 234-base pair consensus gamma satellite monomer. In Northern analysis, these satellite DNA sequences hybridized predominantly to an approximately 1.8-kb RNA species in polyadenylated RNA from P19 cells. These transcripts were strongly repressed by retinoic acid, and nuclear run-on assays revealed that this repression was, at least in part, mediated at the transcriptional level. Satellite transcripts were also detected in HeLa cells, where they were similarly down-regulated by retinoids. Heterogeneously sized satellite transcripts were detected in RNA from specific mouse tissues, such as fetuses (but not placenta), adult liver, and testis. In situ hybridization analysis revealed that satellite transcripts are generated from opposite DNA strands and are differentially expressed in cells of the developing central nervous system as well as in adult liver and testis. These data may have implications on retinoic acid-mediated transcriptional regulation and centromere function.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">High frequency DNA rearrangements associated with mouse centromeric satellite DNA.</div>
   <div class="article_authors">
    <span class="article_year">1986</span>
    
      <span class="article_author">Butner KA</span>
    
      <span class="article_author">Lo CW</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">A DNA transformed mouse cell line, generated by the microinjection of a pBR322 plasmid containing the herpes thymidine kinase (tk) gene, was observed to exhibit a high frequency of DNA rearrangement at the site of exogenous DNA integration. The instability in this cell line does not appear to be mediated by the tk inserts or the immediately adjacent mouse DNA, but instead may be a consequence of the larger host environment at the chromosomal site of tk insertion. Results obtained from restriction analysis, in situ chromosome hybridizations, and cesium chloride density-gradient fractionations indicate that the tk inserts are organized as a single cluster of direct and inverted repeats embedded within pericentromeric satellite DNA. To determine the molecular identity of the flanking host sequences, one of the mouse-tk junction fragments was cloned, and subsequent restriction and sequence analyses revealed that this DNA fragment consists almost entirely of classical mouse satellite DNA. On the basis of these observations, we suggest that the instability in this cell line may reflect the endogenous instability or fluidity of satellite DNA.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Microfluorometric detection of asymmetry in the centromeric region of mouse chromosomes.</div>
   <div class="article_authors">
    <span class="article_year">1974</span>
    
      <span class="article_author">Lin MS</span>
    
      <span class="article_author">Latt SA</span>
    
      <span class="article_author">Davidson RL</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Correlation between dinucleotide periodicities and nucleosome positioning on mouse satellite DNA.</div>
   <div class="article_authors">
    <span class="article_year">1990</span>
    
      <span class="article_author">Lowman H</span>
    
      <span class="article_author">Bina M</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Previous studies demonstrated 16 well-defined nucleosome locations (A-P) on a tandemly repeated prototype 234 base pair (bp) mouse satellite repeat unit. We have aligned the A-P fragments to search for DNA sequence elements that might contribute to nucleosome placement at these positions. Our results demonstrate a strikingly regular, uninterrupted, periodic pattern for the AA dinucleotide occurrences along the entire length of the aligned fragments. The periodicity of the AA occurrences is about 9.7 bp. The pattern exhibits a local minimum at position 74, near the nucleosome dyad axis of symmetry. Other dinucleotides--including AC: GT, CA: TG, and CC: GG--are also placed periodically, but their patterns of occurrence are less regular and less frequent than AA. The calculated spacings between consecutive preferred nucleosome locations on mouse satellite DNA are nearly identical, corresponding to multiples of 9.7 bp. The correlation between the periodicity of dinucleotide occurrences and the average spacing of nucleosome positions suggests that the preferred nucleosome locations recur at intervals that may correspond to the DNA helical repeat in the mouse satellite nucleosomes, and that the histone octamers sample (or slip along) the duplex in steps of 9.7 bp during nucleosome formation on mouse satellite DNA.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Mouse centric and pericentric satellite repeats form distinct functional heterochromatin.</div>
   <div class="article_authors">
    <span class="article_year">2004</span>
    
      <span class="article_author">Guenatri M</span>
    
      <span class="article_author">Bailly D</span>
    
      <span class="article_author">Maison C</span>
    
      <span class="article_author">Almouzni G</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Heterochromatin is thought to play a critical role for centromeric function. However, the respective contributions of the distinct repetitive sequences found in these regions, such as minor and major satellites in the mouse, have remained largely unsolved. We show that these centric and pericentric repeats on the chromosomes have distinct heterochromatic characteristics in the nucleus. Major satellites from different chromosomes form clusters associated with heterochromatin protein 1alpha, whereas minor satellites are individual entities associated with centromeric proteins. Both regions contain methylated histone H3 (Me-K9 H3) but show different micrococcal nuclease sensitivities. A dinucleosome repeating unit is found specifically associated with major satellites. These domains replicate asynchronously, and chromatid cohesion is sustained for a longer time in major satellites compared with minor satellites. Such prolonged cohesion in major satellites is lost in the absence of Suv39h histone methyltransferases. Thus, we define functionally independent centromeric subdomains, which spatio-temporal isolation is proposed to be important for centromeric cohesion and dissociation during chromosome segregation.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">The fluorescence localization of mouse satellite DNA in interphase nuclei.</div>
   <div class="article_authors">
    <span class="article_year">1976</span>
    
      <span class="article_author">Mayfield JE</span>
    
      <span class="article_author">Ellison JR</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Nuclei from various mouse tissues exhibit a pattern of fluorescence characteristic of the cell type when stained with the fluorescent compound Hoechst 33258. When such preparations are hybridized in situ with 3H-RNA complementary to the A-T rich satellite of mouse, it is clearly seen that only the fluorescent regions of the nuclei contain the satellite DNA. Thus Hoechst 33258 allows the precise localization of satellite DNA at all stages of the mouse cell cycle.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Transcription of satellite DNA in mouse L-cells.</div>
   <div class="article_authors">
    <span class="article_year">1973</span>
    
      <span class="article_author">Cohen AK</span>
    
      <span class="article_author">Huh TY</span>
    
      <span class="article_author">Helleiner CW</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
 <tr>
  <td>
  <div class="article"> 
   <div class="article_title">Something from nothing: the evolution and utility of satellite repeats.</div>
   <div class="article_authors">
    <span class="article_year">1998</span>
    
      <span class="article_author">Csink AK</span>
    
      <span class="article_author">Henikoff S</span>
    
   </div>
   <div class="article_journal"></div>
   <div class="article_volume"></div>
   <div class="article_pages"></div>
   
     <div class="article_abstract">Large blocks of tandemly repeated sequences, or satellites, surround the centromeres of complex eukaryotes. During mitosis in Drosophila, satellite DNA binds proteins that, during interphase, bind other sites. The requirement for a repeat to borrow a partner protein from those available at mitosis might limit the spectrum of repeat units that can be expanded into large blocks. To account for the ubiquity and pericentric localization of satellites, we propose that they are utilized to maintain regions of late replication, thus ensuring that the centromere is the last region to replicate on a chromosome.</div>
   
  </div>
  </td>
  <td>
   <div class="article_keywords">
    
   </div> 
  </td>
 </tr>
 
</table>